app/uk/gov/hmrc/personaldetailsvalidation/formats/PersonalDetailsFormat.scala

```diff
@@ -30,6 +30,7 @@ import scala.util.Try
 object PersonalDetailsFormat {
   implicit val withNinoFormats: Format[PersonalDetailsWithNino] = Json.format[PersonalDetailsWithNino]
   implicit val withPostCodeFormats: Format[PersonalDetailsWithPostCode] = Json.format[PersonalDetailsWithPostCode]
+  private val postCodeValidation = """^([A-Za-z][A-HJ-Ya-hj-y]?[0-9][A-Za-z0-9]?|[A-Za-z][A-HJ-Ya-hj-y][A-Za-z])\s?[0-9][ABDEFGHJLNPQRSTUWXYZabdefghjlnpqrstuwxyz]{2}$""".r
```

Comment:
If we do toUppercase before passing the value to the regex, we can have a slightly less complicated looking regex:

"""^([A-Z][A-HJ-Y]?[0-9][A-Z0-9]?|[A-Z][A-HJ-Y][A-Z])\s?[0-9][ABDEFGHJLNPQRSTUWXYZ]{2}$""".r

Reply:
Do we really want to perform two actions when one will work, even if the regular expression is a little bit more complicated?